--- OTBTargets-release.cmake	2015-06-30 11:34:44.723668765 +0200
+++ OTBTargets-release.cmake	2015-06-30 11:41:05.077293889 +0200
@@ -1322,24 +1322,18 @@
 LIST(APPEND _IMPORT_CHECK_TARGETS otbTestDriver )
 LIST(APPEND _IMPORT_CHECK_FILES_FOR_otbTestDriver "${_IMPORT_PREFIX}/bin/otbTestDriver" )
 
+SET(OTB_TARGETS_MISSING)
 # Loop over all imported files and verify that they actually exist
 FOREACH(target ${_IMPORT_CHECK_TARGETS} )
   FOREACH(file ${_IMPORT_CHECK_FILES_FOR_${target}} )
     IF(NOT EXISTS "${file}" )
-      MESSAGE(FATAL_ERROR "The imported target \"${target}\" references the file
-   \"${file}\"
-but this file does not exist.  Possible reasons include:
-* The file was deleted, renamed, or moved to another location.
-* An install or uninstall procedure did not complete successfully.
-* The installation package was faulty and contained
-   \"${CMAKE_CURRENT_LIST_FILE}\"
-but not all the files it references.
-")
+      LIST(APPEND OTB_TARGETS_MISSING ${target})
     ENDIF()
   ENDFOREACH()
   UNSET(_IMPORT_CHECK_FILES_FOR_${target})
 ENDFOREACH()
 UNSET(_IMPORT_CHECK_TARGETS)
+LIST(REMOVE_DUPLICATES OTB_TARGETS_MISSING)
 
 # Cleanup temporary variables.
 SET(_IMPORT_PREFIX)
